vnext/Microsoft.ReactNative/Fabric/Composition/CompositionDynamicAutomationProvider.cpp

```diff
@@ -327,4 +333,62 @@ HRESULT __stdcall CompositionDynamicAutomationProvider::Invoke() {
   return S_OK;
 }
 
-} // namespace winrt::Microsoft::ReactNative::implementation
+std::string convertLPCWSTRToString(LPCWSTR val) {
+  int strLength = WideCharToMultiByte(CP_UTF8, 0, val, -1, nullptr, 0, nullptr, nullptr);
+  std::string str(strLength, 0);
+  WideCharToMultiByte(CP_UTF8, 0, val, -1, &str[0], strLength, nullptr, nullptr);
+  return str;
+}
+
+BSTR StringToBSTR(const std::string &str) {
+  // Calculate the required BSTR size in bytes
+  int len = MultiByteToWideChar(CP_UTF8, 0, str.c_str(), -1, nullptr, 0);
+  if (len == 0) {
+    return nullptr; // Conversion error
+  }
+
+  // Allocate memory for the BSTR
+  BSTR bstr = SysAllocStringLen(nullptr, len - 1); // len includes the null terminator
+
+  // Convert the std::string to BSTR
+  MultiByteToWideChar(CP_UTF8, 0, str.c_str(), -1, bstr, len);
+
+  return bstr;
+}
+
+HRESULT __stdcall CompositionDynamicAutomationProvider::SetValue(LPCWSTR val) {
+  auto strongView = m_view.view();
+
+  if (!strongView)
+    return UIA_E_ELEMENTNOTAVAILABLE;
+
+  auto props = std::static_pointer_cast<const facebook::react::ViewProps>(strongView->props());
+  std::string value = convertLPCWSTRToString(val);
+  auto baseView = std::static_pointer_cast<::Microsoft::ReactNative::CompositionBaseComponentView>(strongView);
+  if (baseView == nullptr)
+    return UIA_E_ELEMENTNOTAVAILABLE;
+  winrt::IInspectable uiaProvider = baseView->EnsureUiaProvider();
+  UpdateUiaProperty(uiaProvider, UIA_ValueValuePropertyId, *props->accessibilityValue.text, value);
```

Comment:
setValue should go to the ComponentView, which would decide if it needs to update its state based on the new value.
Then it should be up to the ComponentView to call  UpdateUiaProperty(uiaProvider, UIA_ValueValuePropertyId, *props->accessibilityValue.text, value); when the value changes.  Which it would also need to do when the value changes outside of a call to the AutomationPovider::SetValue.

Reply:
Could you clarify what you mean by going to ComponentView? I tried adding an `UpdateUiaProperty()` call in `updateAccessibilityProps()` which was getting called every time the accessibilityValue prop was updated, but there was no change in the UIA properties on insights.  

Also. not sure if this is expected behaviour, but SetValue() never seemed to be getting called when updating accessibilityValue (breakpoint in the function was never hit). My guess was that the accessibilityValue example in accesible.tsx doesn't call SetValue() to update the property.

Reply:
The call to SetValue on the AutomationProvider would only occur if the accessibility tool wants to manipulate the value.  This would happen say through voice control or if you select the value pattern actions button in insights and set the value from there.

That value set should then go to the component view to actually modify its value.

Separately, whenever the component view updates its value it needs to notify the automation peer that its value has been changed.  That should happen through a call to UpdateUiaProperty, which would notify the automation provider of the new value, and then the automation provider should call UiaRaiseAutomationEvent which will notify say insights that the value has changed, at which point it can decide if it needs the updates value, and it would call back into the automation peer's getvalue.